<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MesonConfigPage</class>
 <widget class="QWidget" name="MesonConfigPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>587</width>
    <height>499</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QComboBox" name="i_buildDirs">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="b_addDir">
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset theme="list-add">
         <normaloff>.</normaloff>.</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="b_rmDir">
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset theme="edit-delete">
         <normaloff>.</normaloff>.</iconset>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="c_01_basic">
     <property name="title">
      <string>Basic configuration</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="l_buildType">
        <property name="text">
         <string>Build type:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="KComboBox" name="i_buildType"/>
      </item>
      <item row="1" column="1">
       <widget class="KUrlRequester" name="i_installPrefix">
        <property name="mode">
         <set>KFile::Directory|KFile::ExistingOnly|KFile::LocalOnly</set>
        </property>
        <property name="placeholderText">
         <string>Full path to the installation prefix</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="l_installPrefix">
        <property name="text">
         <string>Installation prefix:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="c_02_buildConfig">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="title">
      <string>Build configuration</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QScrollArea" name="scrollArea">
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="scrollAreaWidgetContents">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>555</width>
           <height>267</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="l_TODO">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>TODO -- Implement in seperate UI file</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="MesonAdvancedSettings" name="advanced" native="true"/>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KComboBox</class>
   <extends>QComboBox</extends>
   <header>kcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>KUrlRequester</class>
   <extends>QWidget</extends>
   <header>kurlrequester.h</header>
  </customwidget>
  <customwidget>
   <class>MesonAdvancedSettings</class>
   <extends>QWidget</extends>
   <header>settings/mesonadvancedsettings.h</header>
   <container>1</container>
   <slots>
    <signal>configChanged()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>b_addDir</sender>
   <signal>clicked()</signal>
   <receiver>MesonConfigPage</receiver>
   <slot>addBuildDir()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>529</x>
     <y>22</y>
    </hint>
    <hint type="destinationlabel">
     <x>295</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>b_rmDir</sender>
   <signal>clicked()</signal>
   <receiver>MesonConfigPage</receiver>
   <slot>removeBuildDir()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>567</x>
     <y>22</y>
    </hint>
    <hint type="destinationlabel">
     <x>295</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>i_buildDirs</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MesonConfigPage</receiver>
   <slot>changeBuildDirIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>22</y>
    </hint>
    <hint type="destinationlabel">
     <x>295</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>addBuildDir()</slot>
  <slot>removeBuildDir()</slot>
  <slot>changeBuildDirIndex(int)</slot>
 </slots>
</ui>
